web/crux/src/app/deploy/deploy.http.controller.ts

```diff
@@ -1,52 +1,174 @@
-import { Controller, Post, Body, Get, UseGuards, UseInterceptors, UseFilters } from '@nestjs/common'
-import { ApiBody, ApiCreatedResponse, ApiOkResponse } from '@nestjs/swagger'
-import HttpExceptionFilter from 'src/filters/http-exception.filter'
-import { Empty } from 'src/grpc/protobuf/proto/common'
-import { CreateDeploymentRequest, CreateEntityResponse, IdRequest } from 'src/grpc/protobuf/proto/crux'
-import HttpLoggerInterceptor from 'src/interceptors/http.logger.interceptor'
-import PrismaErrorInterceptor from 'src/interceptors/prisma-error-interceptor'
 import {
-  CreateDeploymentRequestDto,
-  CreateEntityResponseDto,
-  IdRequestDto,
-  DeploymentEventsDto,
-} from 'src/swagger/crux.dto'
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  Param,
+  Patch,
+  Post,
+  Query,
+  UseGuards,
+  UseInterceptors,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common'
+import { ApiBody, ApiCreatedResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger'
 import { Identity } from '@ory/kratos-client'
+import HttpLoggerInterceptor from 'src/interceptors/http.logger.interceptor'
+import PrismaErrorInterceptor from 'src/interceptors/prisma-error-interceptor'
+import { CreatedResponse, CreatedWithLocation } from '../shared/created-with-location.decorator'
+import CreatedWithLocationInterceptor from '../shared/created-with-location.interceptor'
 import JwtAuthGuard, { IdentityFromRequest } from '../token/jwt-auth.guard'
+import {
+  CreateDeploymentDto,
+  DeploymentDetailsDto,
+  DeploymentDto,
+  DeploymentEventDto,
+  InstanceDto,
+  InstanceSecretsDto,
+  PatchDeploymentDto,
+  PatchInstanceDto,
+} from './deploy.dto'
 import DeployService from './deploy.service'
-import DeployStartValidationPipe from './pipes/deploy.start.pipe'
+import DeployCreateTeamAccessGuard from './guards/deploy.create.team-access.guard'
+import DeployTeamAccessGuard from './guards/deploy.team-access.guard'
+import DeployCopyValidationInterceptor from './interceptors/deploy.copy.interceptor'
+import DeployCreateValidationInterceptor from './interceptors/deploy.create.interceptor'
+import DeleteDeploymentValidationInterceptor from './interceptors/deploy.delete.interceptor'
+import DeployPatchValidationInterceptor from './interceptors/deploy.patch.interceptor'
+import DeployStartValidationInterceptor from './interceptors/deploy.start.interceptor'
+
+const DeploymentId = () => Param('deploymentId')
+const InstanceId = () => Param('instanceId')
 
-@UseGuards(JwtAuthGuard)
-@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor)
-@UseFilters(HttpExceptionFilter)
-@Controller('deploy')
+const ROUTE_DEPLOYMENTS = 'deployments'
+const ROUTE_DEPLOYMENT_ID = ':deploymentId'
+const ROUTE_INSTANCES = 'instances'
+const ROUTE_INSTANCE_ID = ':instanceId'
```

Comment:
Am I the only one who is missing a new line here?

Reply:
No, me too, just the gitlense got me confused. nice catch. :D